src/main/java/com/iexec/sms/secret/compute/TeeTaskComputeSecretService.java

```diff
@@ -65,9 +66,10 @@ public Optional<TeeTaskComputeSecret> getSecret(
         }
         final TeeTaskComputeSecret secret = oSecret.get();
         final String decryptedValue = encryptionService.decrypt(secret.getValue());
-        secret.setValue(decryptedValue);
-
-        return oSecret;
+        // deep copy to avoid altering original object
+        TeeTaskComputeSecret decryptedSecret = SerializationUtils.clone(secret);
```

Comment:
[From the doc](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/SerializationUtils.html#clone-T-):
```
This is many times slower than writing clone methods by hand on all objects in your object graph. However, for complex object graphs, or for those that don't support deep cloning this can be a simple alternative implementation.
```
Are we sure we want to use that method?

Reply:
You raise a good point. Deep cloning brings safety but we don't need it here.
All fields are either primitive types or Enum types in this case.
It should work with `secret.clone()` but if we add a complex type later, we won't have deep cloning.
Sould we add a `clone()` method which returns `this.clone()` in `TeeTaskComputeSecret` ?
If we do it manually, we really need to add tests to guarantee we have different objects with identical values.

Reply:
If the goal is to bring a fix quickly, I agree with this fix.
We can improve the implementation afterwards.

Reply:
Thanks for heads-up, as discussed, I added a todo note: https://github.com/iExecBlockchainComputing/iexec-sms/pull/107/commits/208df368cd8453a0a4a521aae3a43499303de3b4, to be reworked in next versions. Is it fine for you?

Reply:
For further reference
https://dzone.com/articles/shallow-and-deep-java-cloning